### ISSUE TITLE
Support subclasses

### ISSUE THREAD

**Issue by Zalastax** [BlackEdder/painlessjson#8]
I don't think we can handle subclasses right now.
This is how Jackson solves it: http://wiki.fasterxml.com/JacksonPolymorphicDeserialization
I'm not sure how this should be implemented since we use CTFE. My idea is to generate code for all know sub-classes. I don't know how we find those sub-classes. Another problem is that when a sub-class is added all modules that use painlessjson will have to be recompiled.

**Comment by BlackEdder** [comment]
Good point. Also can't think of a better way of doing it. Would be nice to support, not sure how to do it though. Did you happen to look whether cerealed supports it?
